include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -846,6 +846,16 @@ class charge_distribution_surface<Lyt, false> : public Lyt
 
         return std::nullopt;
     }
+    /**
+     * This function allows the local electrostatic potential for some given index position to be set externally.
+     *
+     * @param index The index defining the SiDB position.
+     * @param loc_pot The local electrostatic potential to assign to the given index position (unit: V).
+     */
+    void assign_local_potential_by_index(const uint64_t index, const double loc_pot) noexcept
+    {
```

Comment:
maybe ``.at()``?

Reply:
This is also a mutating operation. I'll put an assert tho.